### ISSUE TITLE
PANIC: Expected record selection to contain required model ID fields.: ConversionFailure("\"b87e42f3-1dc0-4d46-ac41-2c0e9e602241\"", "Int") in query-engine\core\src\interpreter\interpreter.rs:68:26

### ISSUE THREAD

**Issue by WorkAttackDev** [prisma/prisma#13238]
Hi Prisma Team! My Prisma Client just crashed. This is the report:
## Versions

| Name            | Version            |
|-----------------|--------------------|
| Node            | v16.14.2           | 
| OS              | windows            |
| Prisma Client   | 3.13.0             |
| Query Engine    | 0.1.0              |
| Database        | mysql              |

## Logs
```
  prisma:tryLoadEnv Environment variables loaded from C:\wa\salu-io\.env  
  prisma:tryLoadEnv Environment variables loaded from C:\wa\salu-io\.env  
  prisma:client dirname  C:\wa\salu-io\node_modules\.prisma\client  
  prisma:client relativePath  ..\..\..\prisma  
  prisma:client cwd  C:\wa\salu-io\prisma  
  prisma:client clientVersion: 3.13.0  
  prisma:client clientEngineType: library  
  prisma:client:libraryEngine internalSetup  
  prisma:client:libraryEngine Searching for Query Engine Library in C:\wa\salu-io\node_modules\.prisma\client  
  prisma:client:libraryEngine loadEngine using C:\wa\salu-io\node_modules\.prisma\client\query_engine-windows.dll.node  
  prisma:client:libraryEngine sending request, this.libraryStarted: false  
  prisma:client:libraryEngine sending request, this.libraryStarted: false  
  prisma:client:libraryEngine library starting  
  prisma:client:libraryEngine library already starting, this.libraryStarted: false  
  prisma:client:libraryEngine library started  +3s
  prisma:client:libraryEngine sending request, this.libraryStarted: true  +2s
  prisma:client:libraryEngine sending request, this.libraryStarted: true  +4s
  prisma:client:libraryEngine sending request, this.libraryStarted: true  
  prisma:client:libraryEngine sending request, this.libraryStarted: true  
  prisma:client:libraryEngine sending request, this.libraryStarted: true  
  prisma:client:libraryEngine sending request, this.libraryStarted: true  
  prisma:client:libraryEngine sending request, this.libraryStarted: true  
  prisma:client:libraryEngine sending request, this.libraryStarted: true  
  prisma:client:libraryEngine sending request, this.libraryStarted: true  +1s
  prisma:client:libraryEngine sending request, this.libraryStarted: true  
  prisma:client:libraryEngine sending request, this.libraryStarted: true  
  prisma:client:libraryEngine sending request, this.libraryStarted: true  
  prisma:client:libraryEngine sending request, this.libraryStarted: true  
  prisma:client:libraryEngine sending request, this.libraryStarted: true  +2s
  prisma:client:libraryEngine sending request, this.libraryStarted: true  
  prisma:client:libraryEngine sending request, this.libraryStarted: true  
  prisma:client:libraryEngine sending request, this.libraryStarted: true  +3s
  prisma:client:libraryEngine sending request, this.libraryStarted: true  
  prisma:client:libraryEngine sending request, this.libraryStarted: true  
  prisma:client:libraryEngine sending request, this.libraryStarted: true  +5s
  prisma:client:request_handler Error: Expected record selection to contain required model ID fields.: ConversionFailure("\"b87e42f3-1dc0-4d46-ac41-2c0e9e602241\"", "Int")
  prisma:client:request_handler     at prismaGraphQLToJSError (C:\wa\salu-io\node_modules\@prisma\client\runtime\index.js:39695:10)
  prisma:client:request_handler     at Object.request (C:\wa\salu-io\node_modules\@prisma\client\runtime\index.js:41559:17)
  prisma:client:request_handler     at async Object.request (C:\wa\salu-io\node_modules\@prisma\client\runtime\index.js:45599:24)
  prisma:client:request_handler     at async PrismaClient._request (C:\wa\salu-io\node_modules\@prisma\client\runtime\index.js:46456:18)
  prisma:client:request_handler     at async editTaskController (webpack-internal:///(api)/./features/api/task/controllers/editTaskController.ts:38:33)
  prisma:client:request_handler     at async handler (webpack-internal:///(api)/./pages/api/tasks/edit/index.ts:26:17)
  prisma:client:request_handler     at async Object.apiResolver (C:\wa\salu-io\node_modules\next\dist\server\api-utils\node.js:185:9)
  prisma:client:request_handler     at async DevServer.runApi (C:\wa\salu-io\node_modules\next\dist\server\next-server.js:395:9)
  prisma:client:request_handler     at async Object.fn (C:\wa\salu-io\node_modules\next\dist\server\base-server.js:496:37)
  prisma:client:request_handler     at async Router.execute (C:\wa\salu-io\node_modules\next\dist\server\router.js:226:36)  
  prisma:client:libraryEngine sending request, this.libraryStarted: true  +13s
```

## Client Snippet
```ts
// PLEASE FILL YOUR CODE SNIPPET HERE
```

## Schema
```prisma
// PLEASE ADD YOUR SCHEMA HERE IF POSSIBLE
```

## Prisma Engine Query
```
{"X":{}}
```

**Comment by Weakky** [comment]
Hey @WorkAttackDev, would you mind sharing a minimal Prisma Schema for us to reproduce this bug? Thank you 🙏

**Comment by WorkAttackDev** [comment]
Hi @Weakky, Here is the schema:

generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["referentialIntegrity"]
}

datasource db {
  provider             = "mysql"
  url                  = env("DATABASE_URL")
  shadowDatabaseUrl    = env("SHADOW_DATABASE_URL")
  referentialIntegrity = "prisma"
}

// emuns

enum Role {
  USER
  EDITOR
  ADMIN
}

enum TokenType {
  REFRESH_TOKEN
  VERIFICATION_TOKEN
  FORGET_TOKEN
}

enum ProjectStatus {
  TODO
  IN_PROGRESS
  DONE
}

// Models

model Token {
  id        Int       @id @default(autoincrement())
  type      TokenType
  token     String    @unique @default(uuid())
  owner     User      @relation(fields: [userId], references: [id], onDelete: Cascade, onUpdate: Cascade)
  userId    Int
  expiresAt DateTime
  createdAt DateTime  @default(now())
  @@map("tokens")
}

model User {
  id            Int      @id @default(autoincrement())
  email         String   @unique @db.VarChar(255)
  emailVerified Boolean  @default(false)
  name          String   @db.VarChar(200)
  userName      String   @unique @db.VarChar(200)
  password      String   @db.VarChar(250)
  picture       String?  @db.VarChar(300)
  providerId    String?  @db.VarChar(200)
  role          Role     @default(USER)
  updatedAt     DateTime @updatedAt
  createdAt     DateTime @default(now())
  tokens              Token[]
  Projects            Project[]
  ProjectParticipants ProjectParticipant[]
  @@map("users")
}

model Project {
  id          Int           @id @default(autoincrement())
  name        String        @db.VarChar(200)
  description String?       @default("Sem descrição") @db.VarChar(1000)
  status      ProjectStatus @default(TODO)
  tasks       Task[]
  ownerId Int
  owner   User @relation(fields: [ownerId], references: [id])
  startDate DateTime? @db.DateTime
  endDate   DateTime? @db.DateTime
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  participants ProjectParticipant[]
  labels       Label[]
  @@map("projects")
}

model ProjectParticipant {
  id        Int      @id @default(autoincrement())
  projectId Int
  userId    Int
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  project Project @relation(fields: [projectId], references: [id])
  user    User    @relation(fields: [userId], references: [id])
  @@map("project_participants")
}

model Task {
  id          Int           @id @default(autoincrement())
  name        String        @db.VarChar(200)
  description String?       @default("Sem descrição") @db.VarChar(1000)
  status      ProjectStatus @default(TODO)
  projectId Int
  project   Project @relation(fields: [projectId], references: [id])
  startDate DateTime? @db.DateTime
  endDate   DateTime? @db.DateTime
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  labels Label[]
  todos TaskTodo[]
  @@map("tasks")
}

model TaskTodo {
  id   String  @id @default(uuid())
  text String  @db.VarChar(200)
  done Boolean @default(false)
  taskId Int
  task   Task @relation(fields: [taskId], references: [id], onDelete: Cascade)
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  @@map("task_todos")
}

model Label {
  id    String @id @default(uuid())
  name  String @unique @db.VarChar(25)
  color String @default("#ff811c") @db.VarChar(7)
  projects Project[]
  tasks    Task[]
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  @@map("labels")
}


The code that cases the issue is:

the id come from req.body.id.

    const taskData = {
      description: description || undefined,
      endDate: endDate ? new Date(endDate) : null,
      name,
      projectId,
      status,
      startDate: startDate ? new Date(startDate) : new Date(),
    };


 if (!id) {
        const newTask = await prisma.task.create({
          data: taskData,
        });

        res.status(201).json({ data: newTask, errors: null });
        return;
      }

      const updatedTask = await prisma.task.update({
        where: {
          id: id,
        },
        data: taskData,
      });

      res.status(200).json({ data: updatedTask, errors: null });

**Comment by pantharshit00** [comment]
Sorry for the late reply here, can you please also provide the values that you are passing to these client calls? I wasn't able to reproduce with the values that I used. Also, give this another shot with the latest version.

**Comment by Weakky** [comment]
Hey,

I couldn't reproduce this bug and figured out it was fixed by https://github.com/prisma/prisma-engines/pull/2874 and released in 3.14.

Closing this issue now, please re-open it if you're still experiencing the problem 🙏
